test/suite-functional/api/functional.thenable.spec.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+declare (strict_types = 1); // @codeCoverageIgnore
+
+namespace Recoil;
+
+use Eloquent\Phony\Phony;
+
+context('can be invoked by yielding a thenable', function () {
```

Comment:
In regards to these test names, try to make them read well including the word "it". Perhaps:

`context('when it has a then method', ...)`
`rit('resumes the strand when resolved`, ...)`
`rit('resumes the strand with an exception when rejected`, ...)`

etc